### PR TITLE
[feature](nereids) implement uncheckedCast method in VarcharLiteral

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

Issue Number: noissue

## Problem summary

Implement uncheckedCast on varchar Literal for a temp way to let TimestampArithmetic work.
We should remove these code and do implicit cast in TypeCoercion rule in future.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
@morrySnow

**Review by morrySnow — DISMISSED** [review]

**Comment by morrySnow** [comment]
please add regression test

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/VarcharLiteral.java [hunk 1]
```diff
@@ -63,4 +67,28 @@ public LiteralExpr toLegacyLiteral() {
...
+    protected Expression uncheckedCastTo(DataType targetType) throws AnalysisException {
+        if (getDataType().equals(targetType)) {
+            return this;
+        }
+        if (targetType.isDateType()) {
+            return convertToDate(targetType);
+        } else if (targetType.isIntType()) {
+            return new IntegerLiteral(Integer.parseInt(value));
+        }
+        //todo other target type cast
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/VarcharLiteral.java:81]: remove this TODO, and add a comment to this function: temporary way to process type coercion in TimestampArithmetic, should be replaced by TypeCoercion rule.

**Comment by Kikyou1997** [comment]
> please add regression test

added unit test

**Review by englefly — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**